crates/wasi/src/preview2/filesystem.rs

```diff
@@ -109,14 +111,21 @@ pub struct Dir {
     pub dir: Arc<cap_std::fs::Dir>,
     pub perms: DirPerms,
     pub file_perms: FilePerms,
+    pub open_mode: FilePerms,
```

Comment:
Mind adding some brief comments here about what these fields are? Three sets of permissions seems like a good threshold for needing disambiguation